src/main/daml/Daml/Finance/Interface/Util/Disclosure.daml

```diff
@@ -84,16 +84,16 @@ interface Disclosure where
     do
       assertMsg "controller must be authorized to RemoveObservers when removing other parties"
         $ disclosers == snd observersToRemove || disclosers `S.isSubsetOf` (view this).disclosureControllers
-
       let
         key = fst observersToRemove
         curObservers = (view this).observers
-
       case M.lookup key curObservers of
         Some parties -> do
           let
             updatedParties = parties `S.difference` snd observersToRemove
-            newObservers = M.insert key updatedParties curObservers
+            newObservers = if (S.null updatedParties)
+              then M.delete key curObservers
+              else M.insert key updatedParties curObservers
```

Comment:
I presume this means we're missing tests for this scenario...

Reply:
added a test